internal/env/types.go

```diff
@@ -161,3 +162,15 @@ func (t *Float) UnmarshalTOML(v interface{}) error {
 	*t = Float(*f)
 	return nil
 }
+
+type URL url.URL
+
+func (t *URL) UnmarshalTOML(v any) error {
+	u, err := ParseURL(v)
+	if err != nil {
+		return err
+	}
+
+	*t = URL(*u)
```

Comment:
is `nil` in this case unmarshalled as `""`?

Reply:
well, nil in this case would cause a panic... I have pushed up a commit with a fix. Thanks for detailed code review!